src/App.jsx

```diff
@@ -4,36 +4,51 @@ import SpotifyLogin from "./SpotifyLogin";
 import SpotifyPlaylists from "./SpotifyPlaylists";
 import SpotifySongSearch from "./SpotifySongSearch";
 import { getToken, setToken } from "./utils/tokenStorage";
-import { getAccessTokenFromLocationHash } from "./utils/spotify";
+import { getAccessTokenFromLocationHash, fetchUser } from "./utils/spotify";
 import "./css/App.css";
+import UserContext from "./contexts/UserContext";
 
 const App = () => {
   const [authorized, setAuthorized] = useState(false);
+  const [userData, setUserData] = useState(null);
 
   useEffect(() => {
-    if (!getToken()) {
-      const token = getAccessTokenFromLocationHash(window.location.hash);
+    const authorize = async () => {
+      if (!getToken()) {
+        const token = getAccessTokenFromLocationHash(window.location.hash);
+        if (!token) return;
+        setToken(token);
+      }
 
-      if (!token) return;
+      const token = getToken();
+      setAuthorized(true);
 
-      setToken(token);
-    }
+      const user = await fetchUser(`${token}`);
 
-    setAuthorized(true);
+      if (user.hasError) {
+        setAuthorized(false);
+        return;
+      }
+      setUserData({ token, ...user });
+    };
+
+    authorize();
   }, []);
 
   return (
     <Router>
       <Switch>
         <Route path="/">
-          <div className="container">
-            <div className="headContainer">
-              <h1>Spotify playlist generator</h1>
-              <SpotifyLogin authorized={authorized} />
+          <UserContext.Provider value={userData}>
+            <div className="container">
+              <div className="headContainer">
+                <h1>Spotify playlist generator</h1>
+                <SpotifyLogin authorized={authorized} />
+              </div>
+              <SpotifySongSearch authorized={authorized} />
```

Comment:
you can remove authorized prop and replace it with contest.user != null

Reply:
If I remove authorized prop, then I cannot detect when to show loading indicator. I think, I have to give it more suitable name than autjorized.

Reply:
ok, we'll change it later during refactoring